livekit-plugins/livekit-plugins-playht/livekit/plugins/playht/tts.py

```diff
@@ -0,0 +1,190 @@
+    voice: Voice
+    base_url: str
+    sample_rate: int
+
+
+class TTS(tts.TTS):
+    def __init__(
+            self,
+            *,
+            voice: Voice = DEFAULT_VOICE,
+            api_key: str | None = None,
+            user_id: str | None = None,
+            base_url: str | None = None,
+            http_session: aiohttp.ClientSession | None = None,
+    ) -> None:
+        super().__init__(
+            capabilities=tts.TTSCapabilities(
+                streaming=False,
+            ),
+            sample_rate=PLAYHT_TTS_SAMPLE_RATE,
+            num_channels=PLAYHT_TTS_CHANNELS,
+        )
+        api_key = api_key or os.environ.get("PLAYHT_API_KEY")
+        if not api_key:
+            raise ValueError("PLAYHT_API_KEY must be set")
+
+        user_id = user_id or os.environ.get("PLAYHT_USER_ID")
+        if not user_id:
+            raise ValueError("PLAYHT_USER_ID mus be set")
+
+        self._opts = _TTSOptions(
+            voice=voice,
+            user_id=user_id,
+            api_key=api_key,
+            base_url=base_url or API_BASE_URL_V1,
+            sample_rate=self.sample_rate,
+        )
+        self._session = http_session
+
+    def _ensure_session(self) -> aiohttp.ClientSession:
+        if not self._session:
+            self._session = utils.http_context.http_session()
+
+        return self._session
+
+    async def list_voices(self) -> List[Voice]:
+        async with self._ensure_session().get(
+                f"{self._opts.base_url}/voices",
+                headers={
+                    "accept": "application/json",
+                    AUTHORIZATION_HEADER: self._opts.api_key,
+                    USERID_HEADER: self._opts.user_id
+                },
+        ) as resp:
+            return _dict_to_voices_list(await resp.json())
+
+    def synthesize(self, text: str) -> "ChunkedStream":
+        return ChunkedStream(text, self._opts, self._ensure_session())
+
+
+class ChunkedStream(tts.ChunkedStream):
+    """Synthesize using the chunked api endpoint"""
+
+    def __init__(
+            self, text: str, opts: _TTSOptions, session: aiohttp.ClientSession
+    ) -> None:
+        super().__init__()
+        self._text, self._opts, self._session = text, opts, session
+
+    @utils.log_exceptions(logger=logger)
+    async def _main_task(self) -> None:
+        stream = utils.audio.AudioByteStream(
+            sample_rate=self._opts.sample_rate, num_channels=1
+        )
+        request_id = utils.shortuuid()
+        segment_id = utils.shortuuid()
+        parent_path = os.path.dirname(os.path.abspath(__file__))
+        client = Client(self._opts.user_id, self._opts.api_key)
+        options = TTSOptions(
+            voice=self._opts.voice.id,
+            sample_rate=PLAYHT_TTS_SAMPLE_RATE,
+            format=Format.FORMAT_WAV,
+            speed=1
+        )
+
+        try:
+            response = client.tts(text=self._text, voice_engine=self._opts.voice.voice_engine, options=options)
```

Comment:
this request isn't async, it is going to block the event loop

Reply:
I had tested this code and this is synchronous process, it will not block the process.

Reply:
it'd be best to use an [executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor), since this code isn't using async underneath.

Reply:
I replaced the synchronous process with PlayHT's streaming API, so it is now an asynchronous process that doesn't block the event loop.